frontend/src/metabase/admin/settings/components/SettingsPages/EmailSettingsPage.tsx

```diff
@@ -11,19 +12,24 @@ import { AdminSettingInput } from "../widgets/AdminSettingInput";
 import { EmailReplyToWidget } from "../widgets/EmailReplyToWidget";
 
 export function EmailSettingsPage() {
-  const isHosted = useSetting("is-hosted?");
-  const isEmailConfigured = useSetting("email-configured?");
+  const { data: settingValues, isFetching: settingsLoading } =
+    useGetSettingsQuery();
+  const isHosted = settingValues?.["is-hosted?"];
+  const isEmailConfigured = settingValues?.["email-configured?"];
   const hasEmailAllowListFeature = useHasTokenFeature("email_allow_list");
   const hasEmailRestrictRecipientsFeature = useHasTokenFeature(
     "email_restrict_recipients",
   );
 
   const dispatch = useDispatch();
   useEffect(() => {
-    if (!isHosted && !isEmailConfigured) {
+    // it's important to check for settingsLoading to ensure isEmailConfigured is fresh.
+    // Otherwise, /email/smtp form will navigate to /email but isEmailConfigured will still be false
+    // and it'll redirect straight back to /email/smtp
+    if (!isHosted && !isEmailConfigured && !settingsLoading) {
```

Comment:
This is not ideal but I figured best to keep changes to the routing to a minimum while we're rewriting the whole thing